src/Raven.Client/Documents/Changes/DatabaseChanges.cs

```diff
@@ -211,33 +211,22 @@ public IChangesObservable<DocumentChange> ForDocumentsInCollection<TEntity>()
             return ForDocumentsInCollection(collectionName);
         }
 
+        [Obsolete("This method is not supported anymore. Will be removed in next major version of the product.")]
         public IChangesObservable<DocumentChange> ForDocumentsOfType(string typeName)
         {
-            if (typeName == null) throw new ArgumentNullException(nameof(typeName));
-            var encodedTypeName = Uri.EscapeDataString(typeName);
-
-            var counter = GetOrAddConnectionState("types/" + typeName, "watch-type", "unwatch-type", encodedTypeName);
-
-            var taskedObservable = new ChangesObservable<DocumentChange, DatabaseConnectionState>(
-                counter,
-                notification => string.Equals(typeName, notification.TypeName, StringComparison.OrdinalIgnoreCase));
-
-            return taskedObservable;
+            throw new NotSupportedException($"Method {nameof(ForDocumentsOfType)} is obsolete.");
```

Comment:
This is a breaking change for someone who might be calling this.
Might be better to return noop to continue the current behavior.